### PR TITLE
Sort props before other nodes

### PR DESCRIPTION
**Pull request by lexich**
Fix: #23 #29

### CODE REVIEW

**Comment by lexich** [comment]
What do you think about it @Siilwyn ?

**Comment by Siilwyn** [comment]
As [commented by misund](https://github.com/Siilwyn/css-declaration-sorter/issues/23#issuecomment-285501963): "If it's fixed by moving declarations to the top, it might be a blocker for others.". So I'm not sure this is the way to go unless we make it an option. On the other hand options lead to complexity... That said I'll look into this PR, one thing we might want is consistent behaviour so it will be moving all declarations or making them consistent as described in #29.

**Comment by lexich** [comment]
Current behaviuor of sorting isn't feature - it's bug, because comparator can compare rules and props. It's false. If you think that props at the top isn't right way, we should accept this PR and change sorting behaviour for different nodes. But I hope, that props at the top will be by default.

File: src/index.js [hunk 1]
```diff
@@ -100,14 +108,16 @@ module.exports = postcss.plugin('css-declaration-sorter', function (options) {
 
     options = options || {};
 
+    const position = options.position || 'top';
```
Comment (lexich) [inline src/index.js]: I made this functionality with flag. And if you don't like new behaviour by default, we can change `top` to `bottom` but new ordering will be available too.

File: bin/cli-usage.txt [hunk 2]
```diff
@@ -9,6 +9,10 @@ Options:
   --order           Use provided order instead of ordering alphabetically.
   --output          Output to a file instead of writing to the origin or stdout.
   --verbose         Log extra information about the process to the console.
+  --position        Put properties at top of rule of at the bottom. (top by default)
```
Comment (lexich) [inline bin/cli-usage.txt]: I'm not a master of copyright, I believe that text here might be better.

**Comment by Siilwyn** [comment]
So after asking around putting declarations above the selectors is a good standard. I've opened a PR for this PR could you review it please @lexich. The biggest change is of course not having it as an option, besides that some tiny changes to make the code more obvious. I put the helper functions at the bottom because I want move all the 'main' functions (including `module.exports`) to the top.

**Pull request merged**